### DISCUSSION TITLE
What is the Google Photo puzzle percentage? If it never gets to 100%, will I be missing photos?

### DISCUSSION THREAD

**Discussion by jacobstein123** [simulot/immich-go#403] [Q&A]
I ran immich-go on my Google Photos takeout, but the Google Photo puzzle percentage never got above 80%. I noticed that if I ran the program multiple times, a few more files would be uploaded each time, up to a point where no more got uploaded. Even after this, I noticed that many months of photos were missing. 

I redid the takeout and reran immich-go, and now the puzzle percentage is up to 99%. It uploaded another 3000 photos. Running it subsequent times still results in more uploads, so clearly it has not found all the photos.

Is there a way to be sure that all the photos from the takeout were uploaded?

**Comment by simulot** [comment]
Takeout files contain a lot of duplicated files. Hence the completion rate is hard to get right.

I'm working on this issue.

**Comment by ZenomeFR** [comment]
Hi ! I've uploaded a takeout of 15x50Go (750Go - 150 000photos/videos in the preview)  from my Google Photo and it ended at arround 53%. I've launched it again a couple of time after that but it only added like 3000 photos. If I understand right it's normal ? :) Thank you for your support and thanks to the Immich-go devs for their nice work 👍 !

**Comment by simulot** [comment]
Be sure having processed all parts of the takeout at the same time.

The google takeout archive is full of duplicate files, in many kind... and totals get wrong. A fix for that is almost ready.

Some archives are missing bunch of JSON files. This shouldn't happen, and often, getting another takeout fixes the problem.
With the current version images without a JSON are not uploaded, and totals are wrong.
A fix is almost ready, and I'll add the possibility to force the upload of photos without JSON in the event it's impossible to get a correct takeout

So, check if you have processed all parts and wait for the next release of immich-go

**Reply by ZenomeFR** [reply]
I'm checking my photos today and it seems that many of my images with the same name have had their exif mismatched. As I've been using an iPhone for many years, sometimes my file name loops around and comes back to something like “IMG_3557.JPG” which has been used in the past. As a result, I have a lot of photos with incorrect dates and locations :(

Do you know how I can solve this problem?  
Thanks for your help :)

**Reply by simulot** [reply]
The date and location are taken from the JSON file with the same name as the picture, in the same archive folder.
Be sure to use the latest version  of immich-go (0.21)

**Comment by ZenomeFR** [comment]
Ok thanks for the reply :) Yes all the parts of my takeout were downloaded in the same folder + using the instruction "takeout-*.zip" to have them all.

I'll try later with the new version and a new complete takeout but it will take time to export/download since mine is massive.
Thanks again ! 🥇
